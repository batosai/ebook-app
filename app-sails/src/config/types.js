import { PropTypes as T } from 'react';

export const Library = T.shape({
  id: T.string.isRequired,
  name: T.string.isRequired,
});
