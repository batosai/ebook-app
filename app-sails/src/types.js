export const BOOKS_REQUEST = 'books:request';
export const BOOKS_SUCCESS = 'books:success';
export const BOOKS_FAILURE = 'books:failure';

export const BOOK_REQUEST = 'book:request';
export const BOOK_SUCCESS = 'book:success';
export const BOOK_EDIT_REQUEST = 'book:edit:request';
export const BOOK_EDIT_SUCCESS = 'book:edit:success';
export const BOOK_DELETE_REQUEST = 'book:delete:request';
export const BOOK_DELETE_SUCCESS = 'book:delete:success';

export const ASIDE_TOGGLE = 'aside:toggle';

export const COLLECTIONS_REQUEST = 'collections:request';
export const COLLECTIONS_SUCCESS = 'collections:success';
export const COLLECTIONS_FAILURE = 'collections:failure';
export const COLLECTIONS_FILTER = 'collections:filter';

export const COLLECTION_ADD_REQUEST = 'collection:add:request';
export const COLLECTION_EDIT_REQUEST = 'collection:edit:request';
export const COLLECTION_DELETE_REQUEST = 'collection:delete:request';

export const LIBRARIES_REQUEST = 'libraries:request';
export const LIBRARIES_SUCCESS = 'libraries:success';
export const LIBRARIES_FAILURE = 'libraries:failure';

export const LIBRARY_ADD_REQUEST = 'library:add:request';
export const LIBRARY_EDIT_REQUEST = 'library:edit:request';
export const LIBRARY_DELETE_REQUEST = 'library:delete:request';
