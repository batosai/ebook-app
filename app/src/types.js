
export const BOOKS_REQUEST = 'books:request';
export const BOOKS_SUCCESS = 'books:success';
export const BOOKS_FAILURE = 'books:failure';

export const BOOK_DELETE_REQUEST = 'book:delete';
export const BOOK_DELETE_SUCCESS = 'book:delete:success';

export const ASIDE_TOGGLE = 'aside:toggle';

export const COLLECTIONS_REQUEST = 'collections:request';
export const COLLECTIONS_SUCCESS = 'collections:success';
export const COLLECTIONS_FAILURE = 'collections:failure';
export const COLLECTIONS_FILTER  = 'collections:filter';

export const COLLECTION_ADD_REQUEST    = 'collection:add:request';
export const COLLECTION_EDIT_REQUEST    = 'collection:edit:request';
export const COLLECTION_DELETE_REQUEST = 'collection:delete:request';

export const LIBRARIES_REQUEST = 'libraries:request';
export const LIBRARIES_SUCCESS = 'libraries:success';
export const LIBRARIES_FAILURE = 'libraries:failure';

export const LIBRARY_ADD_REQUEST    = 'library:add:request';
export const LIBRARY_EDIT_REQUEST    = 'library:edit:request';
export const LIBRARY_DELETE_REQUEST = 'library:delete:request';
