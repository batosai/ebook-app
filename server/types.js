
module.exports = {
  'ACTION': 'action',
  'BOOKS_REQUEST': 'books:request',
  'BOOKS_SUCCESS': 'books:success',
  'BOOKS_FAILURE': 'books:failure',
  'BOOK_DELETE_REQUEST' : 'book:delete',
  'BOOK_DELETE_SUCCESS' : 'book:delete:success',
  'ASIDE_TOGGLE': 'books:failure',
  'COLLECTIONS_REQUEST': 'collections:request',
  'COLLECTIONS_SUCCESS': 'collections:success',
  'COLLECTIONS_FAILURE': 'collections:failure',
  'LIBRARIES_REQUEST': 'libraries:request',
  'LIBRARIES_SUCCESS': 'libraries:success',
  'LIBRARIES_FAILURE': 'libraries:failure'
};
