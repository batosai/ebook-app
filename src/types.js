import { PropTypes as T } from 'react';

export const Item = T.shape({
  id: T.string.isRequired,
  name: T.string.isRequired,
  image: T.string,
  pageNumber: T.number,
  type: T.string,
  read: T.bool,
  collection: T.string
});
