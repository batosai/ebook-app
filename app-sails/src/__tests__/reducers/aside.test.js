import reducer from '../../reducers/aside';
import * as types from '../../types';

const initialState = {
  open: false,
};

describe('Aside reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toEqual({
      open: false,
    });

    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle ASIDE_TOGGLE', () => {
    expect(
      reducer(initialState, {
        type: types.ASIDE_TOGGLE,
      }),
    ).toEqual({
      open: true,
    });

    expect(
      reducer(initialState, {
        type: types.ASIDE_TOGGLE,
      }),
    ).toMatchSnapshot();

    ////////////

    expect(reducer({ open: true }, { type: types.ASIDE_TOGGLE })).toEqual({
      open: false,
    });

    expect(
      reducer({ open: true }, { type: types.ASIDE_TOGGLE }),
    ).toMatchSnapshot();
  });
});
