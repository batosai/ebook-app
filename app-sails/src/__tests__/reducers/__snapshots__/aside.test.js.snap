exports[`Aside reducer should handle ASIDE_TOGGLE 1`] = `
Object {
  "open": true,
}
`;

exports[`Aside reducer should handle ASIDE_TOGGLE 2`] = `
Object {
  "open": false,
}
`;

exports[`Aside reducer should return the initial state 1`] = `
Object {
  "open": false,
}
`;
